provider/rfc2136/rfc2136.go

```diff
@@ -439,3 +450,18 @@ func chunkBy(slice []*endpoint.Endpoint, chunkSize int) [][]*endpoint.Endpoint {
 
 	return chunks
 }
+
+func findMsgZone(ep *endpoint.Endpoint, m *dns.Msg, r *rfc2136Provider) {
+	for _, zone := range r.zoneNames {
+		if strings.HasSuffix(ep.DNSName, zone) {
+			r.krb5Realm = strings.ToUpper(dns.Fqdn(zone))
```

Comment:
this looks like a rather bad side effect

Reply:
Why?